pkg/sql/create.go

```diff
@@ -947,6 +947,12 @@ func resolveFK(
 			}
 		}
 		if !found {
+			// Avoid unexpected index builds from ALTER TABLE ADD CONSTRAINT.
+			if mode == sqlbase.ConstraintValidity_Unvalidated {
```

Comment:
the mode is passed as an argument to resolveFK, to `validated` when it is called in during `CREATE` and `unvalidated` when called in `ALTER`